### PR TITLE
Add an optional third parameter to redisget() to specify a default value

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
@petems I got the acceptance tests this time :)

**Comment by petems** [comment]
@ghoneycutt Awesome stuff! 😄 

I might ask you to rebase as I was doing some work to update the function docs to work with puppet-strings (see #208), but overall looks good

File: lib/puppet/parser/functions/redisget.rb [hunk 1]
```diff
@@ -3,20 +3,40 @@
...
+      default = args[2]
+      raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{default.class} for String) for arg3 (default)") if default.is_a?(String) == false
+    end
 
-    redis = Redis.new(:url => url)
-    redis.get(key)
+    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{key.class} for String) for arg1 (key)") if key.is_a?(String) == false
+    raise(Puppet::ParseError, "redisget(): Wrong argument type given (#{url.class} for String) for arg2 (url)") if url.is_a?(String) == false
+
+    begin
```
Comment (petems) [inline lib/puppet/parser/functions/redisget.rb]: This logic looks a little confusing to me, could we clean this up somehow?

File: lib/puppet/parser/functions/redisget.rb [hunk 2]
```diff
@@ -3,20 +3,40 @@
...
+      redis = Redis.new(:url => url)
+      k = redis.get(key)
+      if k == nil and defined?(default) != nil
+        default
+      else
+        k
+      end
+    rescue
```
Comment (petems) [inline lib/puppet/parser/functions/redisget.rb]: Is there a way to pick which error to rescue from here?

**Comment by ghoneycutt** [comment]
No problem, I'll rebase once #208 gets merged.

File: lib/puppet/parser/functions/redisget.rb (hunk 2 above)
Comment (petems) [inline lib/puppet/parser/functions/redisget.rb]: Something like 

```
rescue Exception => e
      if default
        debug "Connection to redis failed with #{e}"
        default
      else
        raise(Puppet::Error, "Connection to redis server failed - #{e}")
      end
    end
```

Gives a little more detail for both the happy and sad paths

File: lib/puppet/parser/functions/redisget.rb (hunk 1 above)
Comment (petems) [inline lib/puppet/parser/functions/redisget.rb]: Possibly something like changing `k` to `returned_value`? And is there a way to break out the nesting? Not sure if there's something better it just looks a little confusing IMO

Comment (ghoneycutt) [inline lib/puppet/parser/functions/redisget.rb]: changed the variable name and added the exception message to the output as requested

**Comment by petems** [comment]
Merged in #209

**Pull request closed without merging**